pkg/networking/networking.go

```diff
@@ -52,6 +52,15 @@ type networkImpl struct {
 	logger       *zerolog.Logger
 }
 
+func LogRequest(r *http.Request, logger *zerolog.Logger) {
+	if logger.GetLevel() != zerolog.TraceLevel { // Only log if trace level is enabled
```

Comment:
I'm not sure if we need this check, I made an assumption here.

Reply:
It probably doesn't hurt.